pyrdp/parser/rdp/orders/parse.py

```diff
@@ -0,0 +1,355 @@
+#
+# This file is part of the PyRDP project.
+# Copyright (C) 2019 GoSecure Inc.
+# Licensed under the GPLv3 or later.
+#
+
+"""
+Parse Drawing Orders.
+
+TODO: Log unsupported orders.
+"""
+import logging
+from io import BytesIO
+
+from pyrdp.core.packing import Uint8, Uint16LE
+from pyrdp.pdu.rdp.fastpath import FastPathOrdersEvent
+from pyrdp.enum.orders import DrawingOrderControlFlags as ControlFlags
+from pyrdp.enum.rdp import GlyphSupport
+from pyrdp.pdu.rdp.capability import CapabilityType
+
+from .frontend import GdiFrontend
+
+from .secondary import CacheBitmapV1, CacheColorTable, CacheGlyph, CacheBitmapV2, CacheBrush, CacheBitmapV3
+from .alternate import CreateOffscreenBitmap, SwitchSurface, CreateNineGridBitmap, \
+    StreamBitmapFirst, StreamBitmapNext, GdiPlusFirst, GdiPlusNext, GdiPlusEnd, GdiPlusCacheFirst, \
+    GdiPlusCacheNext, GdiPlusCacheEnd, FrameMarker
+from .primary import PrimaryContext as Context
+
+LOG = logging.getLogger('pyrdp.fastpath.rdp.orders')
```

Comment:
there are constants for this

Reply:
What is the point of having the logger names be constants in an external module? The logger namespace is local to the module by definition, and is not supposed to be used by other modules if the point is to namespace log output. Do we have plans to rename loggers in the future? Do we have plans to re-use the same loggers in multiple areas?

This is a lot of unnecessary fragmentation in my opinion. The furthest I'm willing to concede is to have the top level logger name be a constant.

Reply:
It depends on the logger use case. In this case, its for a "program status" kind of use, so it's fine using module name as logger names. What I meant is that you should use `__name__` instead of `pyrdp.fastpath.rdp.orders`. In this module `__name__` will be equal to `pyrdp.parser.rdp.orders` which give pretty much the same information as in your hardcoded string. It allows easier refactor, as changing names of modules should not require to look in string values in a program (or at least not unnecessarily)

"constants" was probably a poor choice of word, I think the correct term is "magic variable"

Reply:
I had not thought of using `__name__`. That is indeed better than a hardcoded string.